### DISCUSSION TITLE
How do I add env variable to containers.conf?

### DISCUSSION THREAD

**Discussion by jahamilton12121** [containers/podman#14309] [Q&A] [answered]
I am trying to make tmp_dir = "/var/tmp/$USER" but it takes it as a literal string.

**Comment by mheon** [comment, marked as answer]
Using other environment variables in an environment variable won't work with containers.conf - it's the shell that's expanding those, so if Podman is managing the environment variable directly through containers.conf no expansion will happen.

**Reply by jahamilton12121** [reply]
I made a workaround by just editing the conf inside the user directory for each user I make. Still it seems unnecessary when I feel as though editing one file should be enough

**Reply by rhatdan** [reply]
We currently do interpret environment variables for certain fields.  No reason not for this one, but does Podman pay attention to this in rootless mode?

**Reply by jahamilton12121** [reply]
Yes in rootless podman will look here: /usr/share/containers/containers.conf first and then look to the one inside the user directory. There is a third place as well it can look.

**Reply by rhatdan** [reply]
Sure, but the path specified in tmp_dir for /usr/share/containers/containers.conf and /etc/containers/containers.conf might not be writable by rootless users.
